shell/components/__tests__/NamespaceFilter.test.ts

```diff
@@ -0,0 +1,213 @@
+      });
+      const dropdown = wrapper.find(`[data-testid="namespaces-dropdown"]`);
+
+      await dropdown.trigger('click');
+      const menu = wrapper.find(`[data-testid="namespaces-menu"]`);
+
+      expect(menu).toBeDefined();
+    });
+
+    it('should contain no options', async() => {
+      const text = 'no options';
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [],
+          options:  () => [],
+          value:    () => [],
+        },
+        mocks:      { $store: { getters: { 'i18n/t': () => text } } },
+        directives: { shortkey: () => jest.fn() }
+      });
+      const dropdown = wrapper.find(`[data-testid="namespaces-dropdown"]`);
+
+      await dropdown.trigger('click');
+      const option = wrapper.find(`[data-testid="namespaces-option-none"]`).element.textContent;
+
+      expect(option).toContain(text);
+    });
+
+    it('should contain an option', async() => {
+      const text = 'my option';
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [
+            {
+              kind:      'namespace',
+              label:     `default-${ text }`,
+            },
+          ],
+          options:  () => [],
+          value:    () => [],
+        },
+        mocks:      { $store: { getters: { 'i18n/t': () => text } } },
+        directives: { shortkey: () => jest.fn() }
+      });
+      const dropdown = wrapper.find(`[data-testid="namespaces-dropdown"]`);
+
+      await dropdown.trigger('click');
+      const option = wrapper.find(`[data-testid="namespaces-option-0"]`).element.textContent;
+
+      expect(option).toContain(text);
+    });
+
+    it('should set the option as user preference', async() => {
+      const text = 'my option';
+      const key = 'my key';
+      const value = {
+        ids: [text],
+        key
+      };
+      const actionName = 'switchNamespaces';
+      const action = jest.fn();
+
+      jest.spyOn(NamespaceFilter.computed.value, 'get').mockReturnValue([]);
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [
+            {
+              label:     text,
+              key,
+              elementId: text,
+              id:        text,
+              kind:      'namespace',
+            },
+          ],
+          options:        () => [],
+          currentProduct: () => undefined,
+          key:            () => key,
+        },
+        mocks: {
+          $store: {
+            getters:  { 'i18n/t': () => text },
+            dispatch: action
+          }
+        },
+        directives: { shortkey: () => jest.fn() }
+      });
+
+      await wrapper.find(`[data-testid="namespaces-dropdown"]`).trigger('click');
+      await wrapper.find(`[data-testid="namespaces-option-0"]`).trigger('click');
+
+      expect(action).toHaveBeenCalledWith(actionName, value);
+    });
+
+    it.todo('should generate the options based on the Rancher resources');
```

Comment:
Is there a plan for this dangling todo?

Reply:
As defined in the ticket, it must be clarified by the author of the code what is defined here.
We may create a ticket to add documentation and bring information, so can be used in the component. As it applies to pretty much any component which includes models or controls, I would rather avoid it for the next iteration or internal discussions.